File: twitterRapid.py
from flask import Blueprint, jsonify, request
import requests, os

twitter_bp = Blueprint("twitterRapid", __name__)

@twitter_bp.route("/likes", methods=["GET"])
def likes():
    try:
        pid = request.args.get("pid")
        count = request.args.get("count", 10)
        headers = {
            "x-rapidapi-host": os.getenv("RAPIDAPI_HOST"),
            "x-rapidapi-key": os.getenv("RAPIDAPI_KEY"),
        }
        resp = requests.get(
            f"https://{os.getenv('RAPIDAPI_HOST')}/likes",
            headers=headers,
            params={"pid": pid, "count": count}
        )
        return jsonify(resp.json())
    except Exception as e:
        return jsonify({"error": str(e)}), 500

File: twitterRapid.js
import express from "express";
import axios from "axios";
const router = express.Router();

router.get("/likes", async (req, res) => {
  try {
    const { pid, count } = req.query;
    const response = await axios.get(`https://${process.env.RAPIDAPI_HOST}/likes`, {
      headers: {
        "x-rapidapi-host": process.env.RAPIDAPI_HOST,
        "x-rapidapi-key": process.env.RAPIDAPI_KEY,
      },
      params: { pid, count },
    });
    res.json(response.data);
  } catch (error) {
    console.error("❌ Twitter RapidAPI error:", error.message);
    res.status(500).json({ error: "Failed to fetch Twitter likes" });
  }
});

export default router;
